File: SampleCode/makerun/makerun.pl
#!/usr/bin/env perl
use strict;
use warnings;

my $target = shift @ARGV // die "Usage: $0 TARGET [ARGS]\n";

system("make", $target);
if($?) {
    exit 1;
}

$target =~ /\// or $target = "./$target";

exec($target, @ARGV);

File: SampleCode/makerun/makerun.js
#!/usr/bin/env node
var argv = [].concat(process.argv);  // shallow copy
var node = argv.shift();
var myname = argv.shift();
if(!argv.length) {
   console.log("Usage: " + myname + " target [ARGS]");
   process.exit();
}
var target = argv.shift();
console.log("run: make " + target);
// ...
